### PR TITLE
add comprehensive complex() string parsing validation tests

### PR DESCRIPTION
**Pull request by sergillam**
This PR adds extensive validation and edge case testing for the complex() string parsing functionality the implementation of parse_complex_string() was already present but lacked comprehensive test coverage.

### CODE REVIEW

File: regression/python/complex_string_parsing/main.py [hunk 1]
```diff
@@ -108,3 +108,214 @@
...
+# Empty and whitespace-only
+raised = False
+try:
+    complex("")
+except ValueError:
+    raised = True
+assert raised
+
+raised = False
+try:
```
[the remark this answers is not in the document]
Comment (sergillam) [reply regression/python/complex_string_parsing/main.py]: done

File: src/python-frontend/function_call_expr.cpp [hunk 2]
```diff
@@ -1390,12 +1415,11 @@ function_call_expr::extract_string_from_symbol(const symbolt *sym) const
...
-        break;
-
       auto decoded = decode_char(ch);
       if (!decoded)
         return std::nullopt;
+      if (*decoded == '\0')
+        break;
       result += *decoded;
     }
   }
```
[the remark this answers is not in the document]
Comment (sergillam) [reply src/python-frontend/function_call_expr.cpp:1449]: ok

File: src/python-frontend/function_call_expr.cpp [hunk 3]
```diff
@@ -1978,6 +2002,45 @@ exprt function_call_expr::handle_complex() const
...
+              const exprt &cond = sym_val.operands()[0];
+              exprt real_part = if_exprt(
+                cond,
+                from_double(t_real, double_type()),
+                from_double(f_real, double_type()));
+              exprt imag_part = if_exprt(
+                cond,
+                from_double(t_imag, double_type()),
+                from_double(f_imag, double_type()));
+              return make_complex(real_part, imag_part);
```
[the remark this answers is not in the document]
Comment (sergillam) [reply src/python-frontend/function_call_expr.cpp]: ok

File: src/python-frontend/function_call_expr.cpp [hunk 4]
```diff
@@ -1978,6 +2002,45 @@ exprt function_call_expr::handle_complex() const
...
+            }
```
[the remark this answers is not in the document]
Comment (sergillam) [reply src/python-frontend/function_call_expr.cpp:2090]: ok
